onnxruntime/core/optimizer/graph_transformer_mgr.cc

```diff
@@ -2,24 +2,50 @@
 // Licensed under the MIT License.
 
 #include "core/optimizer/graph_transformer_mgr.h"
+#include "core/optimizer/rule_based_graph_transformer.h"
 using namespace onnxruntime;
 using namespace ::onnxruntime::common;
 
 namespace onnxruntime {
 
-Status GraphTransformerManager::ApplyAll(Graph& graph) const {
+common::Status GraphTransformerManager::ApplyTransformers(Graph& graph, std::vector<std::string> providers, const TransformerLevel& level) const {
+  const auto& transformers = level_to_transformer_map_.find(level);
```

Comment:
Why carry the level here? You just need the list of transformers. And inside the loop after you don't use but the transformers->second.